src/store/addressbooks.js

```diff
@@ -101,21 +101,26 @@ const mutations = {
 	 * @param {Object} state
 	 * @param {Object} data
 	 * @param {Object} data.addressbook the addressbook
+	 * @param {String} data.sharee the sharee
+	 * @param {Boolean} data.id id
+	 * @param {Boolean} data.group group
 	 */
-	shareAddressbook(state, addressbook, sharee) {
-		addressbook = state.addressbooks.find(search => search === addressbook)
-		let newSharee = {}
-		sharee.displayname = sharee
-		sharee.writable = false
-		addressbook.shares.append(newSharee)
+	shareAddressbook(state, { addressbook, sharee, id, group }) {
+		addressbook = state.addressbooks.find(search => search.id === addressbook.id)
+		let newSharee = {
+			displayname: sharee,
+			id: id,
+			writeable: false,
+			group: group
```

Comment:
Quick tip, you can just put group since it will be the same as group: group.
But don't bother, it's just a quick knowledge bump ;)